### PR TITLE
Bind mouse buttons/wheel to inputs

### PR DESCRIPTION
**Pull request by hiddenasbestos**
This adds the ability to bind mouse buttons/wheel to any user input.
The primary purpose of this work is to support binding lightgun controls in a future update.

Caveats:
* Input driver modifications only applied to dinput and winraw. Other platform drivers for linux, mac, etc. will ignore mouse buttons until upgraded.
* New localisation strings for mouse button names are all in English.
* Binding left or right mouse button to an action with menu mouse control enabled causes the menu to respond to the button press as well. Need some help fixing that one - perhaps re-implement menu controls as hotkeys now?

### CODE REVIEW

File: input/drivers/dinput.c [hunk 1]
```diff
@@ -267,6 +268,58 @@ static bool dinput_keyboard_pressed(struct dinput_input *di, unsigned key)
...
+
+	/* the driver only supports one mouse */
+	if ( settings->uints.input_mouse_index[ port ] != 0 ) {
+		return false;
+	}
+
+	switch ( key )
+	{
+
+	case RETRO_DEVICE_ID_MOUSE_LEFT:
```
Comment (fr500) [inline input/drivers/dinput.c]: you're missing indentation here

**Comment by fr500** [comment]
Hmmm...
I don't really understand this... it's binding as a gamepad, how does that help?

**Comment by fr500** [comment]
> - Binding left or right mouse button to an action with menu mouse control enabled causes the menu to respond to the button press as well. Need some help fixing that one - perhaps re-implement menu controls as hotkeys now?

No there is some latching mechanism that prevents this

File: intl/msg_hash_us.h [hunk 2]
```diff
@@ -763,6 +763,24 @@ MSG_HASH(MENU_ENUM_LABEL_VALUE_INPUT_JOYPAD_Y,
       "Y button (left)")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_INPUT_KEY,
       "(Key: %s)")
+MSG_HASH(MENU_ENUM_LABEL_VALUE_INPUT_MOUSE_LEFT,
+      "MOUSE1")
```
Comment (fr500) [inline intl/msg_hash_us.h]: Should be probably Mouse 1 to match Key strings and Button strings

**Comment by hiddenasbestos** [comment]
It'll make more sense shortly! - My next pull request will be to add 
binds for the new lightgun API. (Although maybe someone will find it 
useful to map buttons on a wireless mouse to hotkeys IDK)

**Comment by fr500** [comment]
Yes I understand that part.

But I thought you were gonna have a RETRO_MOUSE device and then bind a physical mouse to that device instead of binding to gamepad buttons.

I guess that's the missing part?

File: intl/msg_hash_us.h (hunk 2 above)
Comment (hiddenasbestos) [reply intl/msg_hash_us.h]: Happy to rename these to fit in better, etc.  
Mouse 1, Mouse 2, ... 
Wheel Up, Wheel Down, Wheel Left, Wheel Right, ?

Comment (fr500) [reply intl/msg_hash_us.h]: Yeah

**Comment by hiddenasbestos** [comment]
I'm not sure I quite understand, so hopefully my pull request #5790 makes things clearer.
I've added 11 new bindable controls for the lightgun and this pull request allows the user to bind mouse buttons to them - which are how inputs come in from lightgun peripherals on PC (or you just want to play with the mouse)

**Pull request merged**